go.mod

```diff
@@ -30,16 +30,28 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/tecbot/gorocksdb v0.0.0-20191217155057-f0fad39f321c
 	go.etcd.io/bbolt v1.3.8
-	go.uber.org/multierr v1.9.0
-	golang.org/x/crypto v0.15.0
+	go.uber.org/multierr v1.10.0
+	go.uber.org/zap v1.24.0
+	// This version of zap/exp is a requirement until we
+	// upgrade to go 1.21 (https://github.com/uber-go/zap/blob/master/exp/CHANGELOG.md)
+	go.uber.org/zap/exp v0.1.0
+	golang.org/x/crypto v0.18.0
 	golang.org/x/mod v0.14.0
-	golang.org/x/net v0.17.0
-	golang.org/x/term v0.14.0
-	golang.org/x/tools v0.13.0
+	golang.org/x/net v0.20.0
+	golang.org/x/term v0.16.0
+	golang.org/x/tools v0.17.0
 	google.golang.org/protobuf v1.31.0
 	gopkg.in/yaml.v3 v3.0.1
 )
 
+require golang.org/x/exp v0.0.0-20240112132812-db7319d0e0e3
+
+require (
+	github.com/kr/text v0.2.0 // indirect
+	go.uber.org/atomic v1.10.0 // indirect
+	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
+)
+
 require (
 	github.com/btcsuite/btcd/btcec/v2 v2.3.2
 	github.com/btcsuite/btcd/chaincfg/chainhash v1.0.2 // indirect
```

Comment:
and move this to indirect; i think you can have a more consistent and conflict-free go-mod.

Reply:
Moved:

[a8e0dce](https://github.com/gnolang/gno/pull/1302/commits/a8e0dcebb6ca39cca36e05c1da296785414efba6)

I've also enabled a go.mod linter to enforce specific versions of the `zap/exp` package, until we upgrade to `go1.21`